kdump-switch/README.md

```diff
@@ -0,0 +1,21 @@
+# kdump feature switch for Container Optimized OS (COS) in Kubernetes
+
+This is a recipe for enabling/disabling the COS kdump feature in Kubernetes.
+
+On a Kubernetes cluster with COS image (>= M73) as the nodes, you can run the following command to enable kdump feature:
+
+```shell
+kubectl apply -f cos-enable-daemonset.yaml
+```
+
+Or to disable kdump feature:
+```shell
+kubectl delete -f cos-enable-daemonset.yaml
+kubectl apply -f cos-disable-daemonset.yaml
+```
+
+This switch does the following:
+
+1. Enable/disable the kdump feature by running "kdump_helper enable"/"kdump_helper disable".
+2. Reboot the machine to update kernel commandline parameter if needed.
```

Comment:
This sounds disruptive. @yguo0905, do you know what's the behavior when a daemonset reboot the node?

Reply:
All nodes will be rebooted at the same time causing user workloads disruption.

Instead, we can ask customer to create a new node-pool with kdump enabled and migrate their workloads to the new pool. This is the pattern we use for disabling hyper-threading. I don't know there's a better way to do so via DaemonSet.

Reply:
Sound good.
Followed [how disable-smt works](https://cloud.google.com/kubernetes-engine/docs/security-bulletins#may-14-2019), and ask customer to create new node pool, then run the DaemonSet, then migrate the workload.